feat: use official juju_bootstrap TG unit (#103)
canonical/maas-terraform-modules: README.md

The key modules contained in this catalog are:

- [MAAS Deploy](./modules/maas-deploy) - Deploys charmed MAAS
- [MAAS Config](./modules/maas-config) - Performs initial configuration of charmed MAAS

> Before MAAS can be deployed, a Juju controller must exist. Bootstrapping that controller is owned by the Juju team and provided by their official [`terraform-juju-controller`](https://github.com/juju/terraform-juju-controller) module. Our example stacks consume it directly, so there is nothing to bootstrap by hand. If you already have an external Juju controller, you can point the stacks at it instead.

Reviewer: This is very early and is introducing the entire repo, so I think we keep it really brief if we have something like this at all, maybe like
```suggestion
Our example stacks also depend on the external [`terraform-juju-controller`](https://github.com/juju/terraform-juju-controller) module to bootstrap a Juju controller.
```

we have existing docs if they want to bootstrap to an existing controller so I don't think we need to mention that immediately

- OpenTofu/Terraform
- Terragrunt

> - Juju - The [official controller module](https://github.com/juju/terraform-juju-controller) that the stacks use to bootstrap a controller expects the `juju` CLI to be available locally (by default at `/snap/juju/current/bin/juju`). Install it with `sudo snap install juju --channel=3.6/stable`.

Reviewer: This hints at version compatibility/incompatibility here by specifying the channel version, which is confusing. Can you remove this and just reference installing the snap?

Reviewer: Also, it's not just the controller module that expects the path. You could just say more generally "The **modules** expect the `juju` CLI to be available locally..." etc
